### PR TITLE
gh-105017: Include CRLF lines in strings and column numbers

### PR DESCRIPTION
**Pull request by mgmacias95**
* Issue: gh-105017

### CODE REVIEW

File: Python/Python-tokenize.c [hunk 1]
```diff
@@ -240,7 +240,11 @@ tokenizeriter_next(tokenizeriterobject *it)
             type = NAME;
         }
         else if (type == NEWLINE) {
-            str = PyUnicode_FromString("\n");
+            if (it->tok->start[0] == '\r') {
+                str = PyUnicode_FromString("\r\n");
```
Comment (pablogsal) [inline Python/Python-tokenize.c:245]: Byw, I think we may be leaking the previous `str` (we are missing a `XDECREF` here)

**Comment by pablogsal** [comment]
There were a bunch of failing tests because we are removing CRLF handling in the regular path that expects the source to be normalized. I pushed a commit that only preserves CRLF in our tokenize extension

File: Lib/test/test_tokenize.py [hunk 2]
```diff
@@ -85,13 +85,33 @@ def test_basic(self):
...
+    NAME       'x'           (1, 0) (1, 1)
+    OP         '='           (1, 2) (1, 3)
+    NUMBER     '1'           (1, 4) (1, 5)
+    OP         '+'           (1, 6) (1, 7)
+    NUMBER     '1'           (2, 0) (2, 1)
+    NEWLINE    '\\r\\n'        (2, 1) (2, 3)
             """)
 
+
+
```
Comment (pablogsal) [inline Lib/test/test_tokenize.py]:
```suggestion
```

File: Python/Python-tokenize.c (hunk 1 above)
Comment (mgmacias95) [reply Python/Python-tokenize.c:245]: fixed: d3d4ff5fd74338b143c04befb4e7214f05b3ff55

**Review by pablogsal — COMMENTED** [review]
LGTM

Great job! ✨

**Pull request merged**